hoomd/hpmc/update.py

```diff
@@ -490,72 +490,56 @@ def N(self):  # noqa: N802 - allow N as a function name
+    *ref_positions*.
 
-        mc = hpmc.integrate.convex_polyhedron(seed=seed);
-        mc.shape_param.set("A", vertices=verts)
-        mc.set_params(d=0.005, a=0.005)
-        lattice = hpmc.compute.lattice_field(mc=mc, position=fcc_lattice,
-                                             k=1000.0);
-        remove_drift = update.remove_drift(mc=mc, external_lattice=lattice,
-                                           period=1000);
+    Examples::
 
+        rd = hpmc.update.RemoveDrift(ref_positions=[(0, 0, 0), (1, 1, 1)],
+                                     trigger=hoomd.trigger.Periodic(100))
     """
 
-    def __init__(self, mc, external_lattice, period=1):
-        # initialize base class
-        # _updater.__init__(self)
-        cls = None
-        if not hoomd.context.current.device.cpp_exec_conf.isCUDAEnabled():
-            if isinstance(mc, integrate.sphere):
-                cls = _hpmc.RemoveDriftUpdaterSphere
-            elif isinstance(mc, integrate.convex_polygon):
-                cls = _hpmc.RemoveDriftUpdaterConvexPolygon
-            elif isinstance(mc, integrate.simple_polygon):
-                cls = _hpmc.RemoveDriftUpdaterSimplePolygon
-            elif isinstance(mc, integrate.convex_polyhedron):
-                cls = _hpmc.RemoveDriftUpdaterConvexPolyhedron
-            elif isinstance(mc, integrate.convex_spheropolyhedron):
-                cls = _hpmc.RemoveDriftUpdaterSpheropolyhedron
-            elif isinstance(mc, integrate.ellipsoid):
-                cls = _hpmc.RemoveDriftUpdaterEllipsoid
-            elif isinstance(mc, integrate.convex_spheropolygon):
-                cls = _hpmc.RemoveDriftUpdaterSpheropolygon
-            elif isinstance(mc, integrate.faceted_sphere):
-                cls = _hpmc.RemoveDriftUpdaterFacetedEllipsoid
-            elif isinstance(mc, integrate.polyhedron):
-                cls = _hpmc.RemoveDriftUpdaterPolyhedron
-            elif isinstance(mc, integrate.sphinx):
-                cls = _hpmc.RemoveDriftUpdaterSphinx
-            elif isinstance(mc, integrate.sphere_union):
-                cls = _hpmc.RemoveDriftUpdaterSphereUnion
-            elif isinstance(mc, integrate.convex_spheropolyhedron_union):
-                cls = _hpmc.RemoveDriftUpdaterConvexPolyhedronUnion
-            elif isinstance(mc, integrate.faceted_ellipsoid_union):
-                cls = _hpmc.RemoveDriftUpdaterFacetedEllipsoidUnion
-            else:
-                hoomd.context.current.device.cpp_msg.error(
-                    "update.remove_drift: Unsupported integrator.\n")
-                raise RuntimeError("Error initializing update.remove_drift")
-        else:
-            raise RuntimeError(
-                "update.remove_drift: Error! GPU not implemented.")
+    def __init__(self, ref_positions, trigger=1):
+        super().__init__(trigger)
+        _param_dict = dict(ref_positions=list(ref_positions), trigger=trigger)
```

Comment:
`trigger` is not needed in the dict here. Also, no validation of the `ref_positions` is being done here.